test/apps/iodemo/io_demo.cc

```diff
@@ -941,24 +1003,28 @@ class DemoServer : public P2pDemoCommon {
 
         BufferIov *iov             = _data_buffers_pool.get();
         IoWriteResponseCallback *w = _callback_pool.get();
+        ConnectionStat &conn_stat  = _conn_stat_map.find(conn->id())->second;
 
         iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
-        w->init(this, conn, msg->sn, iov, &_curr_state.write_count);
-
+        w->init(this, conn, msg->sn, iov,
+                conn_stat.get_completion_counter<IO_WRITE>());
         assert(iov->size() == 1);
 
+        conn_stat.op_started<IO_WRITE>(msg->data_size);
         conn->recv_am_data((*iov)[0].buffer(), (*iov)[0].size(), data_desc, w);
     }
 
     virtual void dispatch_connection_accepted(UcxConnection* conn) {
-        ++_curr_state.active_conns;
+        bool res = _conn_stat_map.insert(
```

Comment:
why would insert fail?

Reply:
in case of duplicates

Reply:
can we print indicative message or assert message?